src/ripple/app/misc/Validations.cpp

```diff
@@ -456,22 +465,27 @@ class ValidationsImp : public Validations
         if (mWriting)
             return;
 
-        mWriting = true;
-        app_.getJobQueue ().addJob (
-            jtWRITE, "Validations::doWrite",
-            [this] (Job&) { doWrite(); });
+        // addCountedJob() may return false (Job not added) at shutdown.
+        mWriting = app_.getJobQueue ().addCountedJob (
```

Comment:
Thank you for the detailed explanation, that was helpful. I rechecked your work and it looks good.

`mWriting` seems to be serving two purposes. One purpose determines if a write job has been dispatched. The other determines if writing to the database is currently in progress. I suppose for clarity it could be broken up into two bools but I am not suggesting you make that change, the current code works.